web/slashes/scrapers.py

```diff
@@ -0,0 +1,218 @@
+                            soup, os.path.dirname(self.url), "doc"
+                        ),
+                        formatting.inplace_convert_code_block_to_code_block,
+                    ]:
+                        converter(description)
+
+                    text_description = formatting.compress_whitespace(description.text)
+                    return f"{formatting.bold(span.text)}\n>{text_description}"
+            return None
+        except Exception as e:
+            current_app.logger.exception(
+                f"Error while trying to find {self.word} {arg}: {e}"
+            )
+            return None
+
+
+class SubmitsCommandHandler(WebScrapingCommandHandler):
+    def __init__(self, *, relink_timeout):
+        super().__init__(relink_timeout=relink_timeout)
+
+        # @JoshK: Should these be arguments to the superclass constructor?
+        self.url = (
+            "https://htcondor.readthedocs.io/en/latest/man-pages/condor_submit.html"
+        )
+        self.word = "submit file command"
+
+    def get_description(self, page_soup, arg):
+        try:
+            start = page_soup.find("div", id="submit-description-file-commands")
+
+            # No, we can't find_all_next("dt") and pass this regex to string,
+            # because some of our dt tags have children, and "text" isn't a
+            # keyword argument for some insane reason (it gets interpreted
+            # as an attribute search).
+            expr = re.compile(f"^{arg}( |$)", re.I)
+
+            # ... lambda?
+            def text_matches(tag):
+                return tag.name == "dt" and expr.search(tag.text)
+
+            dts = start.find_all_next(text_matches)
+
+            whole_description = None
+            for dt in dts:
+                description = dt.find_next("dd")
+                for converter in [
+                    formatting.inplace_convert_em_to_underscores,
+                    formatting.inplace_convert_inline_code_to_backticks,
+                    formatting.inplace_convert_strong_to_stars,
+                    lambda soup: formatting.inplace_convert_internal_links_to_links(
+                        soup, os.path.dirname(self.url), "std.std-ref"
+                    ),
+                    formatting.inplace_convert_code_block_to_code_block,
+                ]:
+                    converter(description)
+
+                for list in description.find_all("ol"):
+                    replacement = "<br>"
+                    for li in list.select("ol > li"):
+                        replacement += f"\u2022 {li.text}<br>"
+                    list.replace_with(replacement)
+
+                for list in description.find_all("ul"):
+                    replacement = "<br>"
+                    for li in list.select("ul > li"):
+                        replacement += f"\u2022 {li.text}<br>"
+                    list.replace_with(replacement)
```

Comment:
Is this general enough to go in `formatting` as `inplace_convert_lists`? Or does the `<br>` replacement scheme screw with that? Perhaps the line ending can be parameterized?

Reply:
I haven't publicized the /submits command in part because I like neither the results of the code, nor the code itself.  Hiding linebreaks from `format.compress_whitespace()` by making using HTML and then converting them back makes that function seem broken.  This code also throws away the numbers in the lists (which is potentially confusing, although maybe easy to fix).

Obviously the scrapers should all share (the paramaterized, if necessary) conversion routines, and in fact I'd been wondering about refactoring the code currently used to call them (since it's almost all the same in all three places).

Reply:
Ok, sounds like we leave it as-is for this PR, then think about future fixes as /submits continues to evolve.